### PR TITLE
Openaire

### PR DESCRIPTION
**Pull request by chreman**
first implementation of openaire-API
* queries for project ID
* reads params from params JSON - params currently have no effect
* builds input data from get_papers query - currently quite scarce, some metadata, text only from title
* returns output_json
* test_openaire.R included

will look at possible richer output from XML/JSON returns

### CODE REVIEW

File: doc/server_config.md [hunk 1]
```diff
@@ -31,6 +31,8 @@ Make sure you have the following packages installed:
   * xml2 **(Version 1.0 or higher)**
   * jaod (Currently, a Github repository only: http://github.com/ropenscilabs/jaod. Install with  devtools.)
   * rbace (Currently, a Github repository only: http://github.com/ropenscilabs/rbace. Install with devtools.)
+  * ropenaire (For VIPER. Currently, a Github repository only: https://github.com/njahn82/ropenaire. Install with devtools.)
```
Comment (pkraker) [inline doc/server_config.md]: This should be changed to Scott's repository for now (before it is merged back into Najko's).

File: doc/server_config.md [hunk 2]
```diff
@@ -57,4 +59,4 @@ Duplicate config.ini in server/preprocessing/conf/ and rename it to config_local
 * snapshot->getsvg_path: Absolute path to getChartSVG.js
 * snapshot->storage_path: Absolute path to the directory, where the snapshots are stored. **Make sure that your webserver has write access to this file and the containing directory.**
 * snapshot->snapshot_php: PHP File responsible for rendering the bubble in a way to be snapshotted. Relative path to general->host
-* snapshot->snapshot_width: Thumbnail width
\ No newline at end of file
+* snapshot->snapshot_width: Thumbnail width
```
Comment (pkraker) [inline doc/server_config.md]: Thanks, but please change to: `Snapshot width`

Comment (chreman) [reply doc/server_config.md]: will do, but this was an erroneous flagging as changed, since I didn't modify anything here

**Comment by chreman** [comment]
>As discussed, the client will send the acronym as the query and the funding stream and grant id as parameters. You will not need the acronym, but it will make reading the database entries much easier. You will need the tuple of funding stream and grant id to create the query, e.g. roa_pubs([funding_stream] = [grant_id])

as far as I understand, the ropenaire package currently searches for fp7 streams only? @sckott

**Comment by chreman** [comment]
temporarily closed while adding more content data

**Comment by chreman** [comment]
* now with much richer content, based on JSON-returns
* text for clustering/naming now consists of title, abstract, subject keywords, authors and year
* a brief test showed that datasets rarely have abstract/keywords, but always(?) doi
* publication results usually have subject keywords/abstract, but rarely a fulltext link to html/pdf, and in some cases no doi

File: server/preprocessing/other-scripts/test/test-openaire.R [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+library(rstudioapi)
+
+options(warn=1)
+
+wd <- dirname(rstudioapi::getActiveDocumentContext()$path)
+
+setwd(wd) #Don't forget to set your working directory
+
+query <- "247153" #args[2]
```
Comment (pkraker) [inline server/preprocessing/other-scripts/test/test-openaire.R]: As discussed, query will be the acronym of the project. The relevant data for retrieving the project (funding stream & grant id) will be sent as parameters `funding_stream` and `grant_id`.

File: server/preprocessing/other-scripts/test/params_openaire.json [hunk 4]
```diff
@@ -0,0 +1,2 @@
+{"call_id":"FP7-PEOPLE-2013-IOF",
```
Comment (pkraker) [inline server/preprocessing/other-scripts/test/params_openaire.json]: As discussed, the relevant data for retrieving the project are funding stream & grant id and will be sent as parameters `funding_stream` and `grant_id`.

**Comment by pkraker** [comment]
Looking good, thanks for the perseverance! Merging now.

**Pull request merged**